src/braket/aws/aws_device.py

```diff
@@ -334,7 +335,8 @@ def _copy_aws_session(
             )
         else:
             boto_session = boto3.Session(region_name=new_region)
-        return AwsSession(boto_session=boto_session, config=config)
+        braket_client = boto_session.client('braket', endpoint_url=client._endpoint.host)
```

Comment:
* It's generally advisable to not rely on private attributes. It might be a better to idea to access this as: `client.meta.endpoint_url`.
* The existing flow [would have constructed](https://github.com/aws/amazon-braket-sdk-python/blob/main/src/braket/aws/aws_session.py#L44) the client with the `config` on #324. In order to avoid any regressions, it might be a good idea to pass that in while constructing the client here.